noir-projects/aztec-nr/aztec/src/oracle/aes128_decrypt.nr

```diff
@@ -2,8 +2,22 @@
 ///
 /// Returns a padded plaintext, of the same size as the input ciphertext.
 /// Note that between 1-16 bytes at the end of the returned plaintext will be pkcs#7 padding.
+///
+/// A reminder on pkcs#7 padding:
+/// PKCS#7 padding standard says to pad a plaintext to the next multiple of 16 bytes
+/// with repeated bytes of the length of the padding.
+///
+///     If the plaintext is 0 bytes:  Pad with 0x10 (16) sixteen times.
+///     If the plaintext is 1 byte:   Pad with 0x0f (15) fifteen times.
+///     If the plaintext is 2 bytes:  Pad with 0x0e (14) fourteen times.
+///     ...
+///     If the plaintext is 15 bytes: Pad with 0x01 one time.
+///
+/// So the final byte always tells you how long the padding is.
+///
 /// It's up to the calling function to identify and remove that padding.
 /// See the tests below for an example of how.
+///
 /// It's up to the calling function to determine whether decryption succeeded or failed.
 /// See the tests below for an example of how.
 unconstrained fn aes128_decrypt_oracle_wrapper<let N: u32>(
```

Comment:
This function should be made `pub` and have the `_wrapper` suffix removed. We have `wrapper` variants when we end up exposing an unconstrained fn (e.g. `random` calls `random_oracle_wrapper` which calls `random_oracle`). This is because constrained functions cannot call oracles, so we need the wrapper. But since we're not constraining anything here we can just remove the middle layer and have the 'wrapper' be what we expose.